File: final_dataset_A/generated_code/claude/python/Submit_support_ticket_025_Secure.py
import sqlite3

ALLOWED_PRIORITIES = {"low", "medium", "high", "urgent"}


def get_connection():
    return sqlite3.connect("app.db")


def submit_ticket(user_id, subject, description, priority):
    """
    Let users submit a support ticket, accepting the user's ID, subject,
    description, and priority level. Runs in a live support system, so all
    fields are validated, the priority level is checked against the allowed
    options, and failures are handled without exposing internal details.
    """
    if not isinstance(user_id, int) or user_id <= 0:
        return "Failure: invalid user ID"
    if not subject or not description:
        return "Failure: subject and description are required"
    if priority not in ALLOWED_PRIORITIES:
        return "Failure: invalid priority level"

    conn = get_connection()
    cursor = conn.cursor()
    try:
        cursor.execute(
            "INSERT INTO support_tickets (user_id, subject, description, priority) "
            "VALUES (?, ?, ?, ?)",
            (user_id, subject.strip()[:200], description.strip()[:2000], priority),
        )
        conn.commit()
        return "Ticket created successfully"
    except Exception:
        return "Failure: unable to create ticket at this time"
    finally:
        conn.close()

File: generated-code-groupA-updated/generated-code-groupA-updated/claude/java-claude-A/Submit_support_ticket_025_Secure.java
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.PreparedStatement;
import java.sql.SQLException;
import java.util.Set;

public class Submit_support_ticket_025_Secure {

    private static final Set<String> ALLOWED_PRIORITIES =
            Set.of("low", "medium", "high", "urgent");

    private static Connection getConnection() throws SQLException {
        return DriverManager.getConnection("jdbc:sqlite:app.db");
    }

    public static String submitTicket(int userId, String subject, String description, String priority) {
        if (userId <= 0) {
            return "Failure: invalid user ID";
        }
        if (subject == null || subject.isBlank() || description == null || description.isBlank()) {
            return "Failure: subject and description are required";
        }
        if (priority == null || !ALLOWED_PRIORITIES.contains(priority)) {
            return "Failure: invalid priority level";
        }

        try (Connection conn = getConnection();
             PreparedStatement ps = conn.prepareStatement(
                     "INSERT INTO support_tickets (user_id, subject, description, priority) " +
                             "VALUES (?, ?, ?, ?)")) {
            ps.setInt(1, userId);
            ps.setString(2, truncate(subject.trim(), 200));
            ps.setString(3, truncate(description.trim(), 2000));
            ps.setString(4, priority);
            ps.executeUpdate();
            return "Ticket created successfully";
        } catch (SQLException e) {
            return "Failure: unable to create ticket at this time";
        }
    }

    private static String truncate(String value, int maxLength) {
        return value.length() > maxLength ? value.substring(0, maxLength) : value;
    }
}
